docs/reference/changelog.rst

```diff
@@ -2,6 +2,20 @@
 Changelog
 *********
 
+4.2.0 (2024-Sep-12)
+-------------------
+
+Application
+===========
+
+- Add a configuration service to unify handling of command line arguments,
+  environment variables, snap configuration, etc.
+- Properly support ``--shell``, ``--shell-after`` and ``--debug`` on the
```

Comment:
This might be better under a "Lifecycle commands" subheader or something?

Reply:
how about just a "Commands" subheader?